libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -0,0 +1,328 @@
+ */
+AP_EFI_Serial_Hirth::AP_EFI_Serial_Hirth(AP_EFI &_frontend) : AP_EFI_Backend(_frontend) {
+    port = AP::serialmanager().find_serial(AP_SerialManager::SerialProtocol_EFI, 0);
+}
+
+
+/**
+ * @brief checks for response from or makes requests to Hirth ECU periodically
+ * 
+ */
+void AP_EFI_Serial_Hirth::update() {
+    bool status = false;
+    uint32_t now = AP_HAL::millis();
+
+    //log the delay in loop interval
+    internal_state.loop_cnt = now - last_loop_ms;
+    last_loop_ms = now;
+
+    if ((port != nullptr)) {
+
+        //waiting for response
+        if (waiting_response) {
+
+            // reset request if not response for SERIAL_WAIT_TIMEOUT-ms
+            if (now - last_response_ms > SERIAL_WAIT_TIMEOUT) {
+                waiting_response = false;
+                port->discard_input();
+                last_response_ms = now;
+                internal_state.ack_fail_cnt++;
+                if(data_send == 1) {internal_state.ack_s1++;}
+                if(data_send == 2) {internal_state.ack_s2++;}
+                if(data_send == 3) {internal_state.ack_s3++;}
+                if(data_send == 5) {internal_state.ack_thr++;}
+            }
+            else{
+
+                num_bytes = port->available();
+
+                // if already requested
+                if (num_bytes >= expected_bytes) {
+                    
+                    // read data from buffer
+                    uint8_t computed_checksum = 0;
+                    computed_checksum += res_data.quantity = port->read();
+                    computed_checksum += res_data.code = port->read();
+
+                    if (res_data.code == req_data.code) {
+                        for (int i = 0; i < (res_data.quantity - QUANTITY_REQUEST_STATUS); i++) {
+                            computed_checksum += raw_data[i] = port->read();
+                        }
+                    }
+
+                    res_data.checksum = port->read();
+                    computed_checksum = computed_checksum % 256;
+                    // computed_checksum = (256 - computed_checksum);
+                    // discard further bytes if checksum is not matching
+                    if (res_data.checksum != (CHECKSUM_MAX - computed_checksum)) {
+                        internal_state.crc_fail_cnt++;
+                        port->discard_input();
+                    }
+                    else {
+                        internal_state.uptime = now - last_uptime;
+                        last_uptime = now;
+                        internal_state.last_updated_ms = now;
+                        decode_data();
+                        copy_to_frontend();
+                    }
+                    waiting_response = false;
+                }
+            }
+        }
+        
+        //sending cmd
+        if(!waiting_response) {
+
+            // get new throttle value
+            new_throttle = (uint16_t)SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
```

Comment:
```suggestion
            const uint16_t new_throttle = (uint16_t)SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
```

Why `uint16_t`?

Reply:
- Hirth takes this as a 16 bit integer
 - Also wanted to avoid this warning, when used as float - warning: comparing floating-point with '==' or '!=' is unsafe [-Wfloat-equal]